packages/ui/src/components/Tooltip.tsx

```diff
@@ -0,0 +1,332 @@
+      {
+        borderBottomColor: pointerColor,
+        transform: [{ rotate: `${triangleAnglesByPlacement[placement]}deg` }],
+        ...trianglePosition,
+      },
+      style,
+    ])}
+  />
+);
+
+export type TooltipPosition =
+  | "top"
+  | "right"
+  | "bottom"
+  | "left"
+  | "overlay"
+  | "inset-bottom-left"
+  | "inset-bottom-right"
+  | "inset-top-left"
+  | "inset-top-right"
+  | "cursor";
+
+export type TooltipProps = {
+  placement?: TooltipPosition;
+  stayOnScreen?: boolean;
+  show: boolean;
+  anchorRef: React.MutableRefObject<View | null>;
+  children: React.ReactNode;
+};
+
+const getPositions = ({
+  placement,
+  anchor,
+  tooltip,
+}: {
+  placement: string;
+  anchor: Size;
+  tooltip: Size;
+}): TooltipState => {
+  switch (placement) {
+    case "top":
+      return {
+        tooltipPosition: {
+          top: -tooltip.height - 16,
+          left: anchor.width / 2 - tooltip.width / 2,
+        },
+        trianglePosition: {
+          top: tooltip.height - 1,
+          left: tooltip.width / 2 - 7.5,
+        },
+      };
+    case "right":
+      return {
+        tooltipPosition: {
+          top: anchor.height / 2 - tooltip.height / 2,
+          left: anchor.width + 15,
+        },
+        trianglePosition: { top: tooltip.height / 2 - 7.5, left: -15 },
+      };
+    case "bottom":
+      return {
+        tooltipPosition: {
+          top: anchor.height + 15,
+          left: anchor.width / 2 - tooltip.width / 2,
+        },
+        trianglePosition: { top: -15, left: tooltip.width / 2 - 7.5 },
+      };
+    case "left":
+      return {
+        tooltipPosition: {
+          top: anchor.height / 2 - tooltip.height / 2,
+          left: -tooltip.width - 16,
+        },
+        trianglePosition: {
+          top: tooltip.height / 2 - 7.5,
+          left: tooltip.width - 1,
+        },
+      };
+    case "overlay":
+      return {
+        tooltipPosition: {
+          top: 0,
+          left: 0,
+          width: anchor.width,
+          height: anchor.height,
+        },
+        trianglePosition: {},
+      };
+    case "inset-bottom-left":
+      return {
+        tooltipPosition: { bottom: 0, left: 0 },
+        trianglePosition: {},
+      };
+    case "inset-top-left":
+      return {
+        tooltipPosition: { top: 0, left: 0 },
+        trianglePosition: {},
+      };
+    case "inset-top-right":
+      return {
+        tooltipPosition: { top: 0, right: 0 },
+        trianglePosition: {},
+      };
+    case "inset-bottom-right":
+      return {
+        tooltipPosition: { bottom: 0, right: 0 },
+        trianglePosition: {},
+      };
+    default:
+      return { tooltipPosition: {}, trianglePosition: {} };
+  }
+};
+
+type UIPosition = {
+  top?: number;
+  bottom?: number;
+  left?: number;
+  right?: number;
+  width?: number;
+  height?: number;
+};
+
+type TooltipState = {
+  tooltipPosition?: UIPosition;
+  trianglePosition?: UIPosition;
+};
+
+type Position = {
```

Comment:
I had to replace `Measure` with these `Position` and `Size` types: many of the errors in this file were due to the approach of having every property be optional. TS couldn't prove that the fields would actually exist at runtime as you expected.